src/lib/registry/logger-registry.js

```diff
@@ -0,0 +1,58 @@
+const Logger = require('../logger');
+const EmptyRegistry = require('./empty-registry');
+const { throwIf } = require('../utils/type-checks');
+
+class LoggerRegistry extends EmptyRegistry {
+    constructor(notifierRegistry) {
+        super();
+        this.notifierRegistry = notifierRegistry;
+    }
+
+    add(name, notifierName) {
+        throwIf.not.string(name, 'name');
+        throwIf.not.string(notifierName, 'notifierName');
+
+        this._registry[name] = notifierName;
+
+        return this;
+    }
+
+    register(name, notifierName) {
+        throwIf.not.string(name, 'name');
+        throwIf.not.string(notifierName, 'notifierName');
+
+
+        const nameUpper = name.toUpperCase();
+        const nameLower = name.toLowerCase();
+
+        this._registry[nameUpper] = notifierName;
+        this._registry[nameLower] = notifierName;
+
+        return this;
+    }
+
+    get(opts, args = {}) {
+        const badArgs = typeof opts !== 'string' && typeof opts !== 'object';
+
+        if (badArgs) throw new Error('Bad arguments for .logger');
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/loggin-js/loggin-js/pullRequest?prid=7629800)
```suggestion
        if (badArgs) {throw new Error('Bad arguments for .logger');}
```